ramanujantools/cmf/cmf.py

```diff
@@ -315,3 +316,66 @@ def limit(  # noqa: F811
         start: Union[dict, type(None)] = None,
     ) -> Limit:
         return self.limit(trajectory, [iterations], start)[0]
+    
+    def delta(
+            self,
+            trajectory: dict,
+            depth: int,
+            start: dict = None,
+            limit: float = None):
+        r"""
+        Calculates the irrationality measure $\delta$ defined, as:
+        $|\frac{p_n}{q_n} - L| = \frac{1}{q_n}^{1+\delta}$
+        for the walk specified by the trajectory.
+        $p_n$ and $q_n$ are respectively defined as the [0,-1] and [1,-1] elements in walk's matrix (see `Limit` class).
+
+        If limit is not specified (i.e, limit is None),
+        then limit is approximated as limit = self.limit(2 * depth)
+    
+        Args:
+            trajectory: the trajectory of the walk.
+            depth: $n$, is the number of trajectory matrices multiplied. The $\ell_1$ distance walked from the start point is `depth * sum(trajectory.values())`.
+            limit: $L$
+        Returns:
+            the delta value as defined above.
+        """
+        if limit is None:
+            # NOTE: opted to multiply the elements of the list below by sum(trajectory.values())
```

Comment:
This is actually a very fair point, we might wanna actually remove the normalization of depths in the `CMF.walk` method.
I think its time we have this discussion in a group meeting.